extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +131,224 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * Sends an Amp Analytics trigger event to a vendor's cross-domain iframe,
+   * or queues the message if the frame is not yet ready to receive messages.
+   * @param {string} request
+   * @param {!Object<string, string>} transportOptions
+   */
+  sendRequestUsingCrossDomainIframe(request, transportOptions) {
+    const frameData = Transport.crossDomainIframes_[transportOptions['iframe']];
+    user().assert(frameData, 'Trying to send message to non-existent frame');
+    frameData.msgQueue.push({senderId: this.id_, request});
+    if (frameData.isReady) {
+      if (frameData.sendTimer == null) {
+        frameData.sendTimer = timerFor(window).delay(() => {
```

Comment:
"Consider initializing as part of the constructor" - Done.
"have the delay callback check if ready" - will do.